lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
```

Comment:
After a bit of thought, yes. I like that it's one line but I'm not sure it's more clear. I had to think through the truth table a bit.

It is odd to have the query method `valid?` being used as a command though. Hmm...